### PR TITLE
Decode a Morse code message

### PR DESCRIPTION
**Pull request by Eccb7**
### Summary
This pull request introduces a Ruby program for decoding Morse code messages. The program includes three methods: `decode_char`, `decode_word`, and `decode`, which allow for the decoding of individual Morse code characters, entire words, and entire messages, respectively. Additionally, it provides a sample usage of the program to decode a message found in an old bottle.

### Changes Made
- Created a Ruby script (`decoder.rb`) containing the three decoding methods.
- Defined a mapping of Morse code characters to their corresponding letters and numbers in the `MORSE_CODE` hash.
- Implemented the `decode_char` method to decode a single Morse code character.
- Implemented the `decode_word` method to decode an entire word in Morse code.
- Implemented the `decode` method to decode the entire message by splitting it into words and calling `decode_word` for each word.
- Added a test case using the provided Morse code message.
- Documented the code with comments to explain the functionality of each method.

### Testing
- Tested the program by decoding the provided Morse code message: ".-   -... --- -..-   ..-. ..- .-.. .-..   --- ..-.   .-. ..- -... .. . ..."

### Usage
1. Clone this repository.
2. Run the Ruby script using an appropriate Ruby environment (e.g., IRB) to decode Morse code messages.

### Checklist
- [x] The code is properly documented with comments.
- [x] The code follows best practices and is DRY (Don't Repeat Yourself).
- [x] The program successfully decodes Morse code messages.
- [x] The code has been tested, and the provided message has been successfully decoded.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,132 @@
-# ruby-morse-decoder
-This project involves creating a Ruby program to decode Morse code messages, enabling the automated translation of Morse code characters and words into readable text. It promotes code reusability and simplifies the process of deciphering Morse code messages.
+<a name="readme-top"></a>
+
+<div align="center">
+  <img src="murple_logo.png" alt="logo" width="140"  height="auto" />
+  <br/>
+
```
Comment (KanzaTahreem) [inline README.md]: - [x] Great work 👏Kindly ensure to fix the broken logo link in the README for a professional documentation. Also, please ensure that you are not using the **template logo**. Optionally, you can use your own logo if you have one or you can simply remove it 🤓⛏

image

**Review by KanzaTahreem — CHANGES_REQUESTED** [review]
# Changes Required 🔁

Hi @Eccb7 & @kkmanuu 

Good job so far 👏! I really appreciate the effort you put in!  **_However, there are few this that still need to be fixed. But you are almost there!_** 💪 🏁

so close


## Highlights 🎯

There are some excellent points that you have made perfectly. Keep up the great work! 👏
- ✅ Message is decoded well
- ✅ Correct Gitflow
- ✅ Descriptive PR 
- ✅ Correct Linters

## Required Changes 🛑

_Please check the comments under the review._

## Optional suggestions ❕

N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (kkmanuu) [reply README.md]: @KanzaTahreem  Done making changes

**Review by MahmoudBakr23 — APPROVED** [review]
Hi Team 😃, 

Your project is complete! There is nothing else to say other than... It's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done on making the required changes from the previous reviewer. 💯 

Cheers, and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**